src/main/scala/castalia/actors/EndpointRequestInterceptor.scala

```diff
@@ -0,0 +1,28 @@
+package castalia.actors
+
+import akka.actor.ActorRef
+import akka.contrib.pattern.ReceivePipeline
+import akka.contrib.pattern.ReceivePipeline.Inner
+import castalia.matcher.RequestMatch
+import castalia.model.Messages._
+import castalia.model.Model.StubConfig
+
+/**
+  * Triggers registration of endpoint stats by intercepting RequestMatch messages
+  * send to the actors extending this trait.
+  */
+trait EndpointRequestInterceptor {
+
+  this: ReceivePipeline =>
+
+  def stubConfig: StubConfig
+  def metricsCollector: ActorRef
+
+  metricsCollector ! EndpointMetricsInit(stubConfig.endpoint)
+
+  pipelineOuter {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Default case not specified.](https://www.codacy.com/app/scala-academy/Castalia/file/1964138201/issues/source?bid=2998213&fileBranchId=3111257#l23)